src/main/kotlin/io/curity/identityserver/plugin/username/authentication/UsernameAuthenticatorRequestHandler.kt

```diff
@@ -59,13 +55,21 @@ class UsernameAuthenticatorRequestHandler(config: UsernameAuthenticatorPluginCon
 
         userPreferencesManager.saveUsername(postRequestModel.username)
 
-        return Optional.of(createAuthenticationResult(postRequestModel.username))
+        return Optional.of(createAuthenticationResult(postRequestModel.username, postRequestModel.register))
     }
 
-    private fun createAuthenticationResult(userName: String) = AuthenticationResult(
+    private fun createAuthenticationResult(userName: String, registrationIndicator: MutableCollection<String>?):
+            AuthenticationResult {
+        var contextAttributes = ContextAttributes.of(Attributes.of(Attribute.of("iat", Date().time)))
+        if (registrationIndicator?.isNotEmpty() == true)
```

Comment:
I suggest adding a boolean getter to the request model and hiding the "string to bool" conversion in there.